### PR TITLE
Feature/Calendar event overlapping

### PR DESCRIPTION
**Pull request by Barde-E**
# Event View Overlapping Logic + Small Event Styling Improvements

## Quick summary
This PR adds proper visual handling for events that overlap in time. When two (or more) events occur in the same time interval, they now appear side-by-side instead of hiding each other.  
A few small styling tweaks were also added to make each event easier to read and more consistent with the app’s visual design.

## Why this change
Until now, overlapping events were drawn on top of each other, making some of them invisible. With this update, users can clearly see all events happening at the same time.  
The small visual adjustments simply improve readability and match the rest of the app’s UI tone.

## Screenshots
Event_Overlap_portrait_mode
Event_Overlap_landscape_mode

> Note: the current 7-day view makes events look very narrow, so overlaps are harder to appreciate. This will be solved in the next PR introducing a view selector (1-day / 5-day / 7-day / monthly).  
For now, landscape mode provides a clearer view.

### Event overlap calculator
A new `EventOverlapCalculator` determines:
- how many events overlap in the same time range,
- each event’s width fraction,
- each event’s horizontal offset fraction.  
This allows us to correctly position overlapping events side-by-side, without collisions.

### Integration in `EventBlock`
`EventBlock` now receives the calculated width and offset and applies them in its layout.  
This ensures each event is drawn at the right horizontal position, with the right width, depending on its overlap group.

### Updated event styling
The event card received small but meaningful improvements:
- Text is now aligned to the left for better readability.
- The time is shown before the participants.
- Participant names are displayed with lower opacity to reduce visual noise.
- A thin border and a very small shadow were added, giving the impression of a clickable, elevated card.

### Unit tests for the overlap calculator
The tests cover:
- events that do not overlap,
- simple two-event overlaps,
- chained overlaps,
- edge cases at boundaries (start/end times).  

They ensure the calculator produces correct width and offset values for all situations.

### UI tests for overlapping events
The UI tests verify that:
- non-overlapping events are full width,
- overlapping events appear side-by-side,
- triple overlaps are displayed with correct proportional widths and offsets.

## Follows-up
- Add a view mode selector (1-day, 5-day, 7-day).
- Hide the top bar and bottom bar in landscape mode for better calendar visibility.

_This description was reviewed by AI_

This PR is related to https://github.com/swent-agendapp/agendapp/issues/271

### CODE REVIEW

File: app/src/androidTest/java/com/android/sample/ui/calendar/CalendarEventsTest.kt [hunk 1]
```diff
@@ -378,6 +378,137 @@ class EventsOverlapTests : BaseEventsTest() {
...
+    scrollUntilVisible(secondTag)
+
+    val firstBounds = compose.onNodeWithTag(firstTag).fetchSemanticsNode().boundsInRoot
+    val secondBounds = compose.onNodeWithTag(secondTag).fetchSemanticsNode().boundsInRoot
+
+    val firstWidth = firstBounds.right - firstBounds.left
+    val secondWidth = secondBounds.right - secondBounds.left
+
+    // Same horizontal position (allowing a small tolerance due to rendering).
+    assertTrue(kotlin.math.abs(firstBounds.left - secondBounds.left) < 1f)
```
Comment (noa-flo) [inline app/src/androidTest/java/com/android/sample/ui/calendar/CalendarEventsTest.kt]: You should put the delat value `1f` into a global constant for the file.

File: app/src/main/java/com/android/sample/ui/calendar/utils/EventOverlapCalculator.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  /**
+   * Calculates layout information for a set of events that will be rendered in the same day column.
+   *
+   * Overlapping events are placed side by side, each taking 1 / N of the column width, where N is
+   * the size of its overlap group.
+   *
+   * @param events List of events in a single day column.
+   * @return Map from Event to its layout information.
+   */
+  fun calculateEventLayouts(events: List<Event>): Map<Event, EventLayout> {
```
[the remark this answers is not in the document]
Comment (Barde-E) [reply app/src/main/java/com/android/sample/ui/calendar/utils/EventOverlapCalculator.kt]: Thanks a lot for this comment, it made me think much more than expected because I realized two things :
1. Easy part : you're right, if there is too much Events at the same time, it becomes hard to read. As you said, when the views 1-day, 5-days, etc will be implemented, the user will switch to the 1-day view (+ use filters).
-> we could add a "...", but as it simply hides information, why not showing all the overlapping Event (like spaghettis) ?
I think that showing every overlapping Event VS showing a "..." is not is not really different, as in both case the information is not readable (and at least, with the spaghettis, the user can have an overview of what it looks like)
2. Tricky part : you made me realized an other very problematic thing, that deserves a paragraph :

When trying different overlapping situations, I found this specific case :
Overlapping_events_problem
In this case, each of these Events only overlap one (or two) other events. But they're _all_ consider as the same "overlapping group" so their width are divided by 5 ! (instead of divided by 2, and alternating between left/right/left like the schéma below)
image

> By the way, this configuration happens in the entreprise of our stakeholder, where there is classes during the whole afternoon and that always overlap some of the other classes.
If we let it like that, the weekly planning will always looks like the screenshot above, which is problematic

That's why I'm currently re-thinking the algorithm of the Event Overlap Calculator

File: app/src/androidTest/java/com/android/sample/ui/calendar/CalendarEventsTest.kt [hunk 3]
```diff
@@ -378,6 +378,137 @@ class EventsOverlapTests : BaseEventsTest() {
...
     assertTrue("Expected at least one visible segment for multi-day event", nodes.isNotEmpty())
   }
+
+  /**
+   * Verifies that two non-overlapping events on the same day are laid out with the same horizontal
+   * position and width (i.e., they use the full column width).
+   */
+  @Test
+  fun calendarGridContent_nonOverlappingEvents_shareSameHorizontalPositionAndWidth() {
+    val monday = LocalDate.now().with(DayOfWeek.MONDAY)
```
[the remark this answers is not in the document]
Comment (Barde-E) [reply app/src/androidTest/java/com/android/sample/ui/calendar/CalendarEventsTest.kt]: Better practice, I agree ! Done in https://github.com/swent-agendapp/agendapp/pull/272/commits/e8622c2582a0e0fb50565c484a2bcd753a91e073
